# CASE: Number of Questions and Sentences in the test dataset

In [1]:
with open("MemN2N-tensorflow-master/data/ptb.test.txt") as f:
    lines = f.readlines()

In [2]:
# load text
filename = 'MemN2N-tensorflow-master/data/ptb.test.txt'
file = open(filename, 'rt')
text = file.read()
file.close()

In [4]:
print(text)

1 Mary is in the school.
2 Bill is in the kitchen.
3 Is Bill in the bedroom? 	no	2
4 Bill journeyed to the bedroom.
5 Fred travelled to the cinema.
6 Is Bill in the bedroom? 	yes	4
7 Fred went back to the park.
8 Bill is either in the school or the office.
9 Is Bill in the park? 	no	8
10 Mary went to the cinema.
11 Julie is either in the school or the office.
12 Is Fred in the park? 	yes	7
13 Julie is either in the park or the school.
14 Bill went back to the office.
15 Is Bill in the office? 	yes	14
1 Fred journeyed to the office.
2 Fred went back to the cinema.
3 Is Fred in the office? 	no	2
4 Julie is either in the school or the office.
5 Julie moved to the cinema.
6 Is Julie in the cinema? 	yes	5
7 Julie is either in the kitchen or the bedroom.
8 Julie journeyed to the school.
9 Is Julie in the school? 	yes	8
10 Bill is either in the bedroom or the school.
11 Bill went to the bedroom.
12 Is Bill in the office? 	no	11
13 Julie journeyed to the park.
14 Mary travelled to the school.


In [6]:
import nltk
nltk.download('punkt')

[nltk_data] Downloading package punkt to C:\Users\Bhanuja
[nltk_data]     Chouhan\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!


True

In [7]:
from nltk.tokenize import sent_tokenize
tokens = sent_tokenize(text)
print(tokens)

['1 Mary is in the school.', '2 Bill is in the kitchen.', '3 Is Bill in the bedroom?', 'no\t2\n4 Bill journeyed to the bedroom.', '5 Fred travelled to the cinema.', '6 Is Bill in the bedroom?', 'yes\t4\n7 Fred went back to the park.', '8 Bill is either in the school or the office.', '9 Is Bill in the park?', 'no\t8\n10 Mary went to the cinema.', '11 Julie is either in the school or the office.', '12 Is Fred in the park?', 'yes\t7\n13 Julie is either in the park or the school.', '14 Bill went back to the office.', '15 Is Bill in the office?', 'yes\t14\n1 Fred journeyed to the office.', '2 Fred went back to the cinema.', '3 Is Fred in the office?', 'no\t2\n4 Julie is either in the school or the office.', '5 Julie moved to the cinema.', '6 Is Julie in the cinema?', 'yes\t5\n7 Julie is either in the kitchen or the bedroom.', '8 Julie journeyed to the school.', '9 Is Julie in the school?', 'yes\t8\n10 Bill is either in the bedroom or the school.', '11 Bill went to the bedroom.', '12 Is Bill

In [8]:
import re

In [14]:
question = [q for q in text if re.search(r'(\?)$',q)]
print(question)

['?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?']


In [16]:
from collections import Counter
c = Counter(question)
print(c)

Counter({'?': 50})


In [17]:
sentence = [q for q in text if re.search(r'(\.)$',q)]
print(sentence)

['.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.', '.']


In [18]:
s = Counter(sentence)
print(s)

Counter({'.': 99})


# Result: There are 50 questions and 99 sentences in this dataset

# CASE: Sentiment Analysis

In [19]:
import nltk
from nltk.sentiment.vader import SentimentIntensityAnalyzer

C:\ProgramData\Miniconda3\lib\site-packages\nltk\twitter\__init__.py:20: UserWarning: The twython library has not been installed. Some functionality from the twitter package will not be available.
  warnings.warn("The twython library has not been installed. "


In [20]:
sentenceAnalyzer = SentimentIntensityAnalyzer()
for sentence in tokens:
    print(sentence)
    ss = sentenceAnalyzer.polarity_scores(sentence)
    for k in ss:
        a = '{0}: {1},'.format(k, ss[k])
        print(a)

1 Mary is in the school.
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
2 Bill is in the kitchen.
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
3 Is Bill in the bedroom?
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
no	2
4 Bill journeyed to the bedroom.
neg: 0.306,
neu: 0.694,
pos: 0.0,
compound: -0.296,
5 Fred travelled to the cinema.
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
6 Is Bill in the bedroom?
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
yes	4
7 Fred went back to the park.
neg: 0.0,
neu: 0.69,
pos: 0.31,
compound: 0.4019,
8 Bill is either in the school or the office.
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
9 Is Bill in the park?
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
no	8
10 Mary went to the cinema.
neg: 0.268,
neu: 0.732,
pos: 0.0,
compound: -0.296,
11 Julie is either in the school or the office.
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
12 Is Fred in the park?
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
yes	7
13 Julie is either in the park or the school.
neg: 0

neg: 0.167,
neu: 0.833,
pos: 0.0,
compound: -0.296,
14 Bill went to the school.
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
15 Is Fred in the school?
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
no	10
1 Fred went to the school.
neg: 0.268,
neu: 0.732,
pos: 0.0,
compound: -0.296,
2 Fred travelled to the kitchen.
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
3 Is Fred in the kitchen?
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
yes	2
4 Fred went back to the school.
neg: 0.0,
neu: 0.69,
pos: 0.31,
compound: 0.4019,
5 Fred moved to the bedroom.
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
6 Is Fred in the school?
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
no	5
7 Mary is in the office.
neg: 0.306,
neu: 0.694,
pos: 0.0,
compound: -0.296,
8 Mary is either in the kitchen or the kitchen.
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
9 Is Mary in the school?
neg: 0.0,
neu: 1.0,
pos: 0.0,
compound: 0.0,
no	8
10 Julie went to the office.
neg: 0.268,
neu: 0.732,
pos: 0.0,
compound: -0.296,
11 Juli

In [35]:
sentenceAnalyzer = SentimentIntensityAnalyzer()
d = {}
list_ss_neg = []
for sentence in tokens:
    #print(sentence)
    list_ss_neg.append(sentenceAnalyzer.polarity_scores(sentence)['neg'])

In [38]:
nonzero_neg_list=[]
for a in list_ss_neg:
    if a!=0.0:
        nonzero_neg_list.append(a)

In [39]:
print(nonzero_neg_list)

[0.306, 0.268, 0.196, 0.239, 0.306, 0.196, 0.196, 0.268, 0.196, 0.216, 0.306, 0.18, 0.18, 0.306, 0.306, 0.18, 0.167, 0.268, 0.306, 0.268, 0.306, 0.268]


In [41]:
from collections import Counter
count_neg = Counter(nonzero_neg_list)
print(count_neg)
sum(count_neg.values())

Counter({0.306: 7, 0.268: 5, 0.196: 4, 0.18: 3, 0.239: 1, 0.216: 1, 0.167: 1})


22

# There are 22 negative questions/responses in the dataset

In [49]:
sentenceAnalyzer = SentimentIntensityAnalyzer()
d = {}
list_ss_pos=[]
for sentence in tokens:
    #print(sentence)
    list_ss_pos.append(sentenceAnalyzer.polarity_scores(sentence)['pos'])

In [51]:
nonzero_pos_list=[]
for b in list_ss_pos:
    if b!=0.0:
        nonzero_pos_list.append(b)

In [52]:
print(nonzero_pos_list)

[0.31, 0.213, 0.31, 0.231, 0.213, 0.31, 0.213, 0.31, 0.351, 0.31, 0.31, 0.351, 0.351, 0.31, 0.31, 0.351, 0.31, 0.31]


In [53]:
from collections import Counter
count_pos = Counter(nonzero_pos_list)
print(count_pos)
sum(count_pos.values())

Counter({0.31: 10, 0.351: 4, 0.213: 3, 0.231: 1})


18

# There are 18 positive questions/responses in the dataset

In [54]:
sentenceAnalyzer = SentimentIntensityAnalyzer()
d = {}
list_ss_neu=[]
for sentence in tokens:
    #print(sentence)
    list_ss_neu.append(sentenceAnalyzer.polarity_scores(sentence)['neu'])

In [55]:
nonzero_neu_list=[]
for b in list_ss_neu:
    if b!=0.0:
        nonzero_neu_list.append(b)

In [56]:
print(nonzero_neu_list)

[1.0, 1.0, 1.0, 0.694, 1.0, 1.0, 0.69, 1.0, 1.0, 0.732, 1.0, 1.0, 0.787, 1.0, 1.0, 0.69, 1.0, 1.0, 0.804, 1.0, 1.0, 0.769, 1.0, 1.0, 0.787, 1.0, 1.0, 0.761, 1.0, 1.0, 0.69, 1.0, 1.0, 0.694, 1.0, 1.0, 0.804, 1.0, 1.0, 1.0, 1.0, 1.0, 0.787, 1.0, 1.0, 0.69, 1.0, 1.0, 0.804, 1.0, 1.0, 0.649, 1.0, 1.0, 0.732, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.804, 1.0, 1.0, 0.69, 1.0, 1.0, 0.784, 1.0, 1.0, 0.69, 1.0, 1.0, 0.694, 1.0, 1.0, 0.649, 1.0, 1.0, 0.82, 1.0, 1.0, 0.82, 1.0, 1.0, 0.694, 1.0, 1.0, 0.649, 1.0, 1.0, 0.694, 1.0, 1.0, 0.69, 1.0, 0.82, 1.0, 1.0, 1.0, 1.0, 1.0, 0.833, 1.0, 1.0, 0.732, 1.0, 1.0, 0.69, 1.0, 1.0, 0.694, 1.0, 1.0, 0.732, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.649, 1.0, 1.0, 0.694, 1.0, 1.0, 0.69, 1.0, 1.0, 1.0, 1.0, 1.0, 0.69, 1.0, 1.0, 0.732, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]


In [57]:
from collections import Counter
count_neu = Counter(nonzero_neu_list)
print(count_neu)
sum(count_neu.values())

Counter({1.0: 110, 0.69: 10, 0.694: 7, 0.732: 5, 0.804: 4, 0.649: 4, 0.787: 3, 0.82: 3, 0.769: 1, 0.761: 1, 0.784: 1, 0.833: 1})


150

# There are 150 positive questions/responses in the dataset

In [58]:
sentenceAnalyzer = SentimentIntensityAnalyzer()
d = {}
list_ss_compound=[]
for sentence in tokens:
    #print(sentence)
    list_ss_compound.append(sentenceAnalyzer.polarity_scores(sentence)['compound'])

In [59]:
nonzero_compound_list=[]
for b in list_ss_compound:
    if b!=0.0:
        nonzero_compound_list.append(b)

In [60]:
print(nonzero_compound_list)

[-0.296, 0.4019, -0.296, 0.4019, 0.4019, -0.296, 0.4019, 0.4019, -0.296, 0.4019, -0.296, -0.296, 0.4019, 0.4019, -0.296, 0.4019, -0.296, -0.296, 0.4019, -0.296, 0.4019, -0.296, 0.4019, -0.296, -0.296, -0.296, 0.4019, -0.296, 0.4019, -0.296, -0.296, -0.296, 0.4019, -0.296, -0.296, 0.4019, -0.296, 0.4019, 0.4019, -0.296]


In [63]:
from collections import Counter
count_compound = Counter(nonzero_compound_list)
print(count_compound)
#sum(count_neu.values())

Counter({-0.296: 22, 0.4019: 18})


# According to the result, the dataset has more negative responses than positive. 22 negative responses and 18 negative to be precise.